assets/js/metrics_live/index.js

```diff
@@ -176,40 +176,124 @@ class CommonMetric {
   }
 }
 
+function tagMeasurement({ x, y, z }) {
+  updateTagSeriesData.call(this, { x, y, z })
+  this.datasets = this.datasets.map((dataset, index) => {
+    if (dataset.key === "|x|") {
+      dataset.data.push(z)
+    } else {
+      handleTagMeasurement(this.tagSummary[x], dataset, { x, y, z })
+    }
+
+    return dataset
+  })
+}
+
+function updateTagSeriesData({ x, y, z }) {
+  let seriesIndex = this.datasets.findIndex(({ key }) => x === key)
+  if (seriesIndex === -1) {
+    addSummaryTagSeries.call(this, x)
```

Comment:
Is there a way to accomplish this without carrying around the implicit `this`? Pulling out instance functions externally increases the complexity of this code significantly. Why not make them functions on `Summary`, etc to remove this complexity?

Reply:
Thanks @davydog187, I have moved the functions into the Summary class and it's cleaner. 👍